pyvrp/cpp/TimeWindowSegment.h

```diff
@@ -87,18 +115,25 @@ TimeWindowSegment TimeWindowSegment::merge(
 #ifdef PYVRP_NO_TIME_WINDOWS
     return {};
 #else
-    auto const arcDuration = durationMatrix(idxLast, other.idxFirst);
-    auto const diff = duration - timeWarp + arcDuration;
-    auto const diffWait = std::max<Duration>(other.twEarly - diff - twLate, 0);
-    auto const diffTw = std::max<Duration>(twEarly + diff - other.twLate, 0);
-
-    return {idxFirst,
-            other.idxLast,
-            duration + other.duration + arcDuration + diffWait,
-            timeWarp + other.timeWarp + diffTw,
-            std::max(other.twEarly - diff, twEarly) - diffWait,
-            std::min(other.twLate - diff, twLate) + diffTw,
-            std::max(releaseTime, other.releaseTime)};
+    using Dur = pyvrp::Duration;
+
+    // arcDuration is the travel duration from our last to other's first client,
+    // and atOther the time at which we arrive at the other's first client.
```

Comment:
Note: `atOther` is relative to the start time of the route so the comment is not technically correct and therefore I'm also not sure if `atOther` is the best name... `diff` was more true to the relative aspect but also vague... maybe `firstToOtherFirstDuration`?
```suggestion
    // and atOther the total duration between starting at our first client to arriving at the other's first client.
```

Reply:
I've updated the comment to reflect this.